test/e2e/federated-service.go

```diff
@@ -308,6 +313,48 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 	})
 })
 
+// verifyCascadingDeletionForService verifies that services are deleted from
+// underlying clusters when orphan dependents is false and they are not
+// deleted when orphan dependents is true.
+func verifyCascadingDeletionForService(clientset *fedclientset.Clientset, clusters map[string]*cluster, orphanDependents *bool, nsName string) {
+	service := createServiceOrFail(clientset, nsName, FederatedServiceName)
+	serviceName := service.Name
+	// Check subclusters if the service was created there.
+	By(fmt.Sprintf("Waiting for service %s to be created in all underlying clusters", serviceName))
```

Comment:
It seems that this code is common to all service creation, and should be factored out, rather than copied here.